src/core/thread/mle.cpp

```diff
@@ -3877,6 +3877,7 @@ void Mle::HandleChildUpdateResponse(const Message &         aMessage,
         {
             mChildUpdateAttempts     = 0;
             mChildUpdateRequestState = kChildUpdateRequestNone;
+            mDataRequestState        = kDataRequestNone;
```

Comment:
I don't follow this one. Can you please explain more? Thanks.

Here is what I see:
- `mDataRequestState` is related to MLE "Network Data" request command (and not MAC data requst or data polls). It remembers whether we are currently waiting to get "Data Response".
- In `HandleChildUpdateResponse()` as we process the received "Child Update Response" we call `HandleLeaderData()` which then checks if the message contains `Tlv::kNetworkData` and parse the info, and then clears the `mDataRequestState` (which happens at the `exit` of `HandleLeaderData()`).

Reply:
well I think You are wright this one is obsolete. Thanks for spotting it.